### DISCUSSION TITLE
Which part of the code handles identifying moves as fishy?

### DISCUSSION THREAD

**Discussion by jrlodge** [kwsch/PKHeX#4197] [Q&A] [answered]
![image](https://github.com/kwsch/PKHeX/assets/81149290/0249c753-8f1b-4c3c-9e57-d6dd2d0b4c60)

I'm working on an API that uses PKHeX as a dependency and I'd like to add a check for whether moves are fishy, in this case, I believe Bestow is given the caution icon in the GUI because it is an event-only move, but I can't find which part of the code identifies a move as fishy. 

I know there's the LegalityAnalysis.Results:

```
CheckResult { Judgement = Valid, Identifier = Encounter, Comment = Able to hatch an Egg at Met Location., Valid = True, Rating = Valid }
CheckResult { Judgement = Valid, Identifier = Evolution, Comment = Valid., Valid = True, Rating = Valid }
CheckResult { Judgement = Valid, Identifier = Nickname, Comment = Nickname does not match another species name., Valid = True, Rating = Valid }
CheckResult { Judgement = Fishy, Identifier = EVs, Comment = All EVs are zero, but leveled above Met Level., Valid = True, Rating = Fishy }
CheckResult { Judgement = Valid, Identifier = Level, Comment = Current level is not below met level., Valid = True, Rating = Valid }
CheckResult { Judgement = Valid, Identifier = Ribbon, Comment = All ribbons accounted for., Valid = True, Rating = Valid }
CheckResult { Judgement = Valid, Identifier = Ability, Comment = Ability matches ability number., Valid = True, Rating = Valid }
CheckResult { Judgement = Valid, Identifier = Ball, Comment = Ball possible for species., Valid = True, Rating = Valid }
CheckResult { Judgement = Valid, Identifier = Form, Comment = Form is Valid., Valid = True, Rating = Valid }
CheckResult { Judgement = Valid, Identifier = Form, Comment = Form argument is valid., Valid = True, Rating = Valid }
CheckResult { Judgement = Fishy, Identifier = Encounter, Comment = Pokémon HOME Transfer Tracker is missing., Valid = True, Rating = Fishy }
```

But there's nothing in here about moves being fishy. I'm looking for whatever part triggers the caution icon in the GUI.

An optional setting to set these warnings to invalidate the Pokemon could potentially be good, as this example, Pikachu in SV is untradeable but is seen as legal by PKHeX.

Any help would be appreciated, thanks.

**Comment by sora10pls** [comment]
That is not a "fishy move", that indicator is there for moves that a Pokémon can have in its moveset in SWSH from past games, but is unusable in SWSH. The implementation in PKHeX matches what Pokémon HOME does.

Starting with BDSP, moves get reset to match the target game's learnset, but PKHeX still stores byte arrays for usable/unusable moves for each given context. In your example, Bestow is not usable in SV, so PKHeX yields the yellow error icon.

**Reply by jrlodge** [reply]
Makes sense, thanks. 

Do you happen to know how I can access these byte arrays for a given Pokemon object? If there's no built-in function for checking if a move is unusable in a given context I should be able to do some manual checks comparing the Pokemon's moves against usable moves.

Is it something to do with this PKHeX.Core/Editing/PKM/LegalMoveInfo.cs ?

I'm working with Pokemon objects created by GetLegalFromTemplate, my API takes showdown sets and returns EKX bytes.

**Reply by kwsch** [reply, marked as answer]
https://github.com/kwsch/PKHeX/blob/aa841aebe99d832839cd65c08a290497ee61bdf9/PKHeX.Core/Moves/MoveInfo.cs#L100
